src/index.ts

```diff
@@ -22,56 +22,43 @@ export interface Traceparent {
  version (2 hex)
 */
 
-const v_size = 1,
-	t_size = 16,
-	p_size = 8;
 
-const v_offset = 0,
-	t_offset = v_size,
-	p_offset = t_offset + t_size,
-	f_offset = p_offset + p_size;
+const trace_id_size = 16,
+	parent_id_size = 8;
+const default_version = "00";
+const sampled_flag = "00";
 
-/*#__INLINE__*/
-const slice = (buf: Uint8Array, offset: number, end?: number): string =>
-	toHEX(buf.slice(offset, end));
 
-const traceparent = (buf: Uint8Array) => ({
-	version: slice(buf, v_offset, t_offset),
-	trace_id: slice(buf, t_offset, p_offset),
-	parent_id: slice(buf, p_offset, f_offset),
-	// TODO: flags should be binary 0b00000000
-	flags: slice(buf, f_offset),
+const traceparent = (version:string, trace_id:string, parent_id: string, flags:string): Traceparent => ({
+	version,
+	trace_id,
+	parent_id,
+	flags,
 
 	child() {
-		const new_buf = new Uint8Array(buf);
-		fill_random(new_buf, p_offset, p_size);
-		return traceparent(new_buf);
+		const new_buf = new Uint8Array(parent_id_size);
+		fill_random(new_buf, 0, parent_id_size);
+		return traceparent(this.version, this.trace_id, toHEX(new_buf), this.flags);
 	},
 
 	toString() {
 		return `${this.version}-${this.trace_id}-${this.parent_id}-${this.flags}`;
 	},
-});
+})
 
 export const make = (): Traceparent => {
-	const buf = new Uint8Array(26);
-	buf[v_offset] = 0; // version
-	buf[f_offset] = 0; // flags
-	fill_random(buf, t_offset, t_size); // trace-id
-	fill_random(buf, p_offset, p_size); // parent-id
-
-	return traceparent(buf);
+	const total_size = trace_id_size + parent_id_size
+	const buf = new Uint8Array(total_size);
+	fill_random(buf, 0, total_size);
```

Comment:
Doing a single random. Not sure how node.js deals with that but in some languages its a syscall and if you can batch it it saves cpu cycles.

Reply:
ya this is so clever actually  ⚡ ⚡